src/bramble-api.js

```diff
@@ -0,0 +1,325 @@
+/*global define, HTMLElement, MessageChannel */
+
+define([
+    // Change this to filer vs. filer.min if you need to debug Filer
+    "thirdparty/filer/dist/filer.min",
+    "thirdparty/MessageChannel/ChannelUtils",
+    "thirdparty/MessageChannel/message_channel"
+], function(Filer, ChannelUtils) {
+    "use strict";    
+
+    // PROD URL for Bramble, which can be changed below
+    var PROD_BRAMBLE_URL = "https://mozillathimblelivepreview.net/bramble/dist/index.html";
+
+    var FilerBuffer = Filer.Buffer;
+    var UUID = ChannelUtils.UUID;
+
+    function parseEventData(data) {
+        try {
+            data = JSON.parse(data);
+            return data || {};
+        } catch(err) {
+            return {};
+        }
+    }
+
+    /**
+     * The `div` is the element, or id of an element to use when creating
+     * the Bramble iframe. All existing contents of this element will be removed:
+     *
+     * var bramble = new Bramble(someDiv);        // expects someDiv to be an existing element
+     * var bramble = new Bramble('#some-div-id'); // selector for element
+     * var bramble = new Bramble();               // will use document.body
+     * 
+     * You can pass various options as well:
+     *
+     * var bramble = new Bramble({...}); // uses document.body with given options
+     * var bramble = new Bramble(elem, {...});
+     *
+     * Options available include:
+     *   url: <String> a URL to use when loading the Bramble iframe (defaults to prod)
+     *   locale: <String> the locale Brackets should use
+     *   extensions: {
+     *       enable: <Array(String)> a list of extensions to enable
+     *       disable: <Array(String)> a list of extensions to disable
+     *   }
+     *   hideUntilReady: <Boolean> whether to hide Bramble until it's fully loaded.
+     *   ready: <Function> a function to be called when Bramble is fully loaded.
+     *   provider: <Filer.FileSystem.providers.*> a provider to use for the fs, defaults to Memory
+     */
+    function Bramble(div, options) {
+        var self = this;
+
+        // Long-running callbacks for fs watch events
+        var watches = {};
+
+        // The channel port for communication with this instance
+        var port;
+
+        // The iframe's window, for postMessage
+        var brambleWindow;
+
+        // Whether to transfer ownership of ArrayBuffers or not
+        var allowArrayBufferTransfer;
+
+        if (typeof div === "object"  && !(div instanceof HTMLElement)) {
+            options = div;
+            div = null;
+        }
+        options = options || {};
+
+        var id = self._id = "bramble-" + UUID.generate();
```

Comment:
This seems a little over-engineered to me. We won't have more than one instance on a page, so couldn't this be hardcoded without the UUID?